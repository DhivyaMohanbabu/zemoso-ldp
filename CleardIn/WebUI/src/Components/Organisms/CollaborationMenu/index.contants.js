const platforms = [
  {
    id: 1,
    name: 'Box',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/e18e8f54-178c-406a-91f2-8b1d83f47cc0.svg',
    isActive: true,
  }, {
    id: 2,
    name: 'Google suite',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/d95b656e-2878-43df-9c4c-382f94892585-3x.png',
    isActive: true,
  }, {
    id: 3,
    name: 'Docusign',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9e69db6f-95ca-4221-b76f-545a010ea986-3x.png',
    isActive: true,
  }, {
    id: 4,
    name: 'Figma',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/55c1d662-ded6-4972-98be-454c2a085e49.svg',
    isActive: true,
  }, {
    id: 5,
    name: 'Okta',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/9c3444bb-6e74-41b1-80b1-e1d812153448.svg',
    isActive: true,
  }, {
    id: 6,
    name: 'Slack',
    image: 'https://cdn.zeplin.io/5fa102170abc4fbe7467eaa5/assets/73eb42dd-a071-48cd-af85-ead7e7c5648e.svg',
    isActive: false,
  },
];

export default platforms;
