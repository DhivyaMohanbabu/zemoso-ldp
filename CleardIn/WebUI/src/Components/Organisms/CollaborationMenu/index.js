/* eslint-disable react/jsx-indent */
import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import {
  Dialog, Grid, IconButton, Typography,
} from '@material-ui/core';
import CloseIcon from '@material-ui/icons/Close';
import PlatformCard from '../../Molecules/PlatformCard';
import platforms from './index.contants';

const useStyles = makeStyles((theme) => ({
  closeButton: {
    fontSize: '30px',
  },
  header: {
    backgroundColor: theme.palette.clearedIn.grey[200],
    margin: '0 0 1.5rem',
    padding: '0.458rem 1.875rem',
    height: '3.75rem',
  },
  body: {
    paddingLeft: '1.85rem',
  },
  overflowHidden: {
    overflow: 'hidden !important',
  },
  cardContainer: {
    padding: '0 1.85rem 1.85rem 0',
  },
}));

const CollaborationMenu = ({ openFlag, handleClose }) => {
  const classes = useStyles();

  return (
    <Dialog open={openFlag} maxWidth="md" className={classes.overflowHidden}>
      <Grid container direction="column" data-testid="platformCard">
        <Grid item container alignItems="center" justify="space-between" className={classes.header}>
          <Grid item>
            <Typography variant="h6" color="textPrimary">
              Choose Collaboration Platforms
            </Typography>
          </Grid>
          <Grid
            item
          >
            <IconButton
              className={classes.closeButton}
              aria-label="close"
              onClick={handleClose}
              data-testid="close"
              size="small"
            >
              <CloseIcon
                className={classes.closeButton}
              />
            </IconButton>
          </Grid>
        </Grid>
        <Grid item container className={classes.body}>
            {platforms.map((platform, index) => (
              <Grid item sm={4} md={3} key={index} className={classes.cardContainer}>
                <PlatformCard {...platform} />
              </Grid>
            ))}
        </Grid>
      </Grid>
    </Dialog>
  );
};

CollaborationMenu.propTypes = {
  openFlag: PropTypes.bool.isRequired,
  handleClose: PropTypes.func,
};

CollaborationMenu.defaultProps = {
  handleClose: undefined,
};

export default CollaborationMenu;
