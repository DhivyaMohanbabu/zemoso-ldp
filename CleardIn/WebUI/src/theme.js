import { grey } from '@material-ui/core/colors';
import { createMuiTheme } from '@material-ui/core/styles';
// import 'typeface-roboto';

const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#1665d8',
      light: '#b0f1f0',
    },
    secondary: {
      main: '#9ea0a5',
      dark: grey,
    },
    error: {
      main: '#ffada5',

    },
    success: {
      main: '#daefa9',
    },
    info: {
      main: grey[900],
      light: '#fdd570',
      dark: grey[600],
    },
    text: {
      primary: '#1c1c21', // '#3e3f42',
      secondary: '#5c5c5c', // #6b6c6f',
    },
    grey: {
      100: '#f8fafb',
      200: '#f2f4f6',
      300: '#e8ecef',
      400: '#98a9bc',
      500: '#778ca2',
    },
    clearedIn: {
      grey: {
        100: '#f6f6f6',
        200: '#eeeeee',
        300: '#cccccc',
        400: '#949494',
        500: '#666666',
      },
      selectedElementBGColor: '#eaf0ff', // rgba(77, 124, 254, 0.12),
      dark: '#5c5c5c',
      bodyBackground: '#fafafa',
      blue: '#1d4cd7',
    },
    relationShip: {
      vendor: '#daefa9',
      customer: '#fdd570',
      partner: '#b0f1f0',
    },
  },
  typography: {
    fontFamily: ['Roboto', 'Rubik', 'sans-serif'].join(','),
    h1: {
      fontSize: '1.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h2: {
      fontSize: '1.625rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h3: {
      fontSize: '1.25rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h4: {
      fontSize: '1.125rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    h5: {
      fontSize: '1rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    h6: {
      fontSize: '1.25rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 1.05,
      letterSpacing: 'normal',
    },
    subtitle1: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    subtitle2: {
      fontSize: '0.938rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    caption: {
      fontSize: '0.875rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
    },
    link: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    body1: {
      fontSize: '0.875rem',
      fontWeight: 'normal',
      fontStyle: 'normal',
      lineHeight: 1.14,
      letterSpacing: 'normal',
    },
    body2: {
      fontFamily: 'Rubik',
      fontSize: '0.875rem',
      fontWeight: 300,
      fontStyle: 'normal',
      lineHeight: 1.5,
      letterSpacing: 'normal',
    },
    button: {
      fontSize: '0.75rem',
      fontWeight: 500,
      fontStyle: 'normal',
      lineHeight: 'normal',
      letterSpacing: 'normal',
      textTransform: 'none',
    },
  },
  overrides: {
    MuiButton: {
      root: {
        maxWidth: '200px',
        display: 'block',
      },
      label: {
        whiteSpace: 'nowrap',
        overflow: 'hidden',
        textOverflow: 'ellipsis',
      },
    },
  },
});

export default theme;
