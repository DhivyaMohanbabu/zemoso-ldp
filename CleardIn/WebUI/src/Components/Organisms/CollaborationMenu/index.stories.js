import React from 'react';
import CollaborationMenu from '.';

export default {
  title: 'organism/CollaborationMenu',
  component: CollaborationMenu,
};

const Template = ({ ...args }) => <CollaborationMenu {...args} />;

export const Primary = Template.bind({});
Primary.args = {
  openFlag: true,
};
