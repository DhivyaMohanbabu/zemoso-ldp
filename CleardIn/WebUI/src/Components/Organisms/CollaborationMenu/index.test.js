import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/core/styles';
import CollaborationMenu from '.';
import theme from '../../../theme';

it('renders CollaborationMenu snapshot correctly', () => {
  const { container } = render(
    <ThemeProvider theme={theme}>
      <CollaborationMenu openFlag="true" />
    </ThemeProvider>,
  );
  expect(container).toMatchSnapshot();
});
