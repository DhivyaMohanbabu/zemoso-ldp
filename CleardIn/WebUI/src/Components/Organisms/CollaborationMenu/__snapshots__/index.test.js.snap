// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders CollaborationMenu snapshot correctly 1`] = `
<div
  aria-hidden="true"
/>
`;
